From 9ccec0e41eb87aa3e2711993518da784d395099c Mon Sep 17 00:00:00 2001
From: Flora Cui <Flora.Cui@amd.com>
Date: Mon, 15 Feb 2016 15:45:59 +0800
Subject: [PATCH 0943/1110] drm/amd/powerplay: add default clockgating handling

This is to workaround regression introduced in
46c34bcb6a15dd85329a39a5e72c62108626acdc. It should be reverted with a
final fix.

Signed-off-by: Flora Cui <Flora.Cui@amd.com>
Reviewed-by: Rex Zhu <Rex.Zhu@amd.com>
---
 drivers/gpu/drm/amd/powerplay/amd_powerplay.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/powerplay/amd_powerplay.c b/drivers/gpu/drm/amd/powerplay/amd_powerplay.c
index 94b7809..32a6a6f 100644
--- a/drivers/gpu/drm/amd/powerplay/amd_powerplay.c
+++ b/drivers/gpu/drm/amd/powerplay/amd_powerplay.c
@@ -188,10 +188,12 @@ static int pp_set_clockgating_state(void *handle,
 
 	hwmgr = ((struct pp_instance *)handle)->hwmgr;
 
-	if (hwmgr == NULL || hwmgr->hwmgr_func == NULL ||
-			hwmgr->hwmgr_func->update_clock_gatings == NULL)
+	if (hwmgr == NULL || hwmgr->hwmgr_func == NULL)
 			return -EINVAL;
 
+	if (hwmgr->hwmgr_func->update_clock_gatings == NULL)
+		return 0;
+
 	if (state == AMD_CG_STATE_UNGATE)
 		pp_state = 0;
 	else
-- 
2.7.4

